module/PowerShellEditorServices/Start-EditorServices.ps1

```diff
@@ -65,14 +65,39 @@ param(
     [switch]
     $ConfirmInstall,
 
+    [Parameter(ParameterSetName="Stdio",Mandatory=$true)]
     [switch]
     $Stdio,
 
+    [Parameter(ParameterSetName="NamedPipe")]
     [string]
     $LanguageServicePipeName = $null,
 
+    [Parameter(ParameterSetName="NamedPipe")]
+    [string]
+    $DebugServicePipeName = $null,
+
+    [Parameter(ParameterSetName="NamedPipeHalfDuplex",Mandatory=$true)]
+    [switch]
+    $SplitInOutPipes,
+
+    [Parameter(ParameterSetName="NamedPipeHalfDuplexParam",Mandatory=$true)]
```

Comment:
> What's the reason for the `"Param"` parameter set? Is it so that you can either set a switch to split the pipe or nominate a pipename yourself?
> 
> I feel like we probably don't need so many different parameter sets -- what is the envisioned usage of each current set?

Right. I wanted to enforce the contract that one could either specify `SplitInOutPipes` Switch or be enforced to specify read/write pipe names directory. Ok, simplified the parameter sets to just one additional set for simplex named pipes.